src/pages/Controllers/ControllerOverview/ControllerOverview.js

```diff
@@ -0,0 +1,73 @@
+import React from "react";
+import { useSelector } from "react-redux";
+
+import {
+  getModelCounts,
+  getGroupedMachinesDataByStatus,
+  getGroupedApplicationsDataByStatus,
+  getGroupedUnitsDataByStatus
+} from "app/selectors";
+import ControllerChart from "../ControllerChart/ControllerChart";
+
+import "./_controller-overview.scss";
+
+export default function ControllersOverview() {
+  const groupedMachinesDataByStatus = useSelector(
+    getGroupedMachinesDataByStatus
+  );
+  const groupedApplicationsDataByStatus = useSelector(
+    getGroupedApplicationsDataByStatus
+  );
+  const groupedUnitsDataByStatus = useSelector(getGroupedUnitsDataByStatus);
+
+  const { machinesCount, applicationCount, unitCount } = useSelector(
```

Comment:
This component is running a lot of loops over the same data extracting different data. It'll be more efficient to count the lengths you are already accessing below.

Also if you're going to derive the counts from the data in the `ControllerChat` component then these values aren't necessary.